lib/db/schemaupdater.go

```diff
@@ -520,13 +467,21 @@ func (db *schemaUpdater) updateSchemato9(prev int) error {
 		return err
 	}
 
-	for folder, meta := range metas {
-		if err := meta.toDB(t, []byte(folder)); err != nil {
-			return err
-		}
-	}
-
 	db.recordTime(indirectGCTimeKey)
 
 	return t.Commit()
 }
+
+// updateSchemato10 makes sure the sequence numbers in the sequence keys match
+// those in the corresponding file entries.
+func (db *schemaUpdater) updateSchemato10(_ int) error {
+	db.gcMut.RLock()
+	defer db.gcMut.RUnlock()
```

Comment:
FWIW the schema updates are for sure guaranteed to run before we start any gc, so this is unnecessary. I think we need to (mentally at least) always be sure that schema updates run before anything else starts touching the database.